src/test/java/io/github/perrymant/moneymaker/ApplicationTest.java

```diff
@@ -6,8 +6,27 @@
 
 public class ApplicationTest {
 
+    private static final String ERROR = "" +
+            "Error: Invalid argument:\n" +
+            "$ moneymaker help\n" +
+            "for more info.";
+    private static final String RANDOMSTRING = "random stuff here";
+    private static final String HELP_MORE_ARGS = "help with more args";
+    private static final String REPORT_MORE_ARGS = "report more args given";
+
     private TestLogger logger = new TestLogger();
     private Application target = new Application(logger);
+    private static final String REPORT = "" +
+            "╔════════════╤══════════════════╤════════╤═════════╤═════════════╗\n" +
+            "║ Time       │ Transaction Type │ Amount │ Balance │ Description ║\n" +
+            "╠════════════╪══════════════════╪════════╪═════════╪═════════════╣\n" +
+            "║ 2018-01-01 │ CREDIT           │ £1.25  │ £1.25   │ Got paid    ║\n" +
```

Comment:
We shouldn't have to update the test every time we add more data, but as it currently stands it will fail for each new transaction we add.

We should therefore give the `Application` a different kind of `Budget` specific to this test, that will return the same String irrespective of how much data we've really got.

Reply:
- I agree to the first point: "We shouldn't have to update the test every time we add more data, but as it currently stands it will fail for each new transaction we add."
- Do I create a new Budget class: perhaps `TestBudget`?
- Apart from having it implement the Budget interface, I'm not quite sure what the other steps would be....

Reply:
Yes to the second point.
The signature of the enforced method should give you a hint, but you get to implement your own version of the `public String report()`. Just return whatever (other than null and "" - something that can't happen by accident), and assert that the logger gets this string when you call `report`. Your `REPORT` constant definitely qualifies, but it can be much simpler too if you want.